detectable/src/main/java/com/synopsys/integration/detectable/detectables/conan/lockfile/ConanLockfileDetectable.java

```diff
@@ -0,0 +1,110 @@
+import com.synopsys.integration.detectable.detectable.result.GivenFileNotFoundDetectableResult;
+import com.synopsys.integration.detectable.detectable.result.PassedDetectableResult;
+import com.synopsys.integration.detectable.extraction.Extraction;
+import com.synopsys.integration.detectable.extraction.ExtractionEnvironment;
+
+@DetectableInfo(language = "C/C++", forge = "conan", requirementsMarkdown = "Files: conan.lock.")
+public class ConanLockfileDetectable extends Detectable {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    public static final String CONANLOCKFILE = "conan.lock";
+    private final FileFinder fileFinder;
+    private final ConanLockfileExtractor conanLockfileExtractor;
+    private final ConanLockfileExtractorOptions conanLockfileExtractorOptions;
+    private File lockfile;
+
+    public ConanLockfileDetectable(DetectableEnvironment environment, FileFinder fileFinder, ConanLockfileExtractor conanLockfileExtractor,
+        ConanLockfileExtractorOptions conanLockfileExtractorOptions) {
+        super(environment);
+        this.fileFinder = fileFinder;
+        this.conanLockfileExtractor = conanLockfileExtractor;
+        this.conanLockfileExtractorOptions = conanLockfileExtractorOptions;
+    }
+
+    @Override
+    public DetectableResult applicable() {
+        if (conanLockfileExtractorOptions.getLockfilePath().isPresent()) {
+            logger.debug("Conan Lockfile detectable applies because user supplied lockfile path {}", conanLockfileExtractorOptions.getLockfilePath().get());
+            return new PassedDetectableResult();
+        }
+        File discoveredLockfile = fileFinder.findFile(environment.getDirectory(), CONANLOCKFILE);
+        if (discoveredLockfile == null) {
+            return new FileNotFoundDetectableResult(CONANLOCKFILE);
+        }
+        logger.debug("Conan Lockfile detectable applies because Detect found the default lockfile {}", discoveredLockfile.getAbsolutePath());
+        lockfile = discoveredLockfile;
+        return new PassedDetectableResult();
+    }
+
+    @Override
+    public DetectableResult extractable() throws DetectableException {
+        if (conanLockfileExtractorOptions.getLockfilePath().isPresent()) {
+            String givenLockfilePath = conanLockfileExtractorOptions.getLockfilePath().get();
+            Optional<File> verifiedLockfile = verifyFile(givenLockfilePath);
+            if (verifiedLockfile.isPresent()) {
+                lockfile = verifiedLockfile.get();
+            } else {
+                return new GivenFileNotFoundDetectableResult(givenLockfilePath);
+            }
+        }
+        return new PassedDetectableResult();
+    }
+
+    @Override
+    public Extraction extract(ExtractionEnvironment extractionEnvironment) throws ExecutableFailedException {
+        return conanLockfileExtractor.extract(lockfile, conanLockfileExtractorOptions);
+    }
+
+    private Optional<File> verifyFile(String filePath) {
```

Comment:
I don't love having this code here. Do we follow this pattern in other detectables? Is this something we want to do for all detectables?

Reply:
I also realized that it was inconsistent with what other detectors do. The file finder checks for existence and not readability. So I took out the readability check, and inlined the little bit of code that remained (the existence check).